### DISCUSSION TITLE
Remote Control and EEPROM on GEN3

### DISCUSSION THREAD

**Discussion by martinpichlo** [wills106/homeassistant-solax-modbus#1121] [General]
Hey there,

first of all: Thank you for for this powerful integration! I use it with Solax X3-Hybrid-6 inverter. 

I have started some experiments with [Modbus Power Control (remotecontrol)](https://homeassistant-solax-modbus.readthedocs.io/en/latest/modbus-power-control/). 

Regarding the documentation the given automations are for Gen4. Are there any reasons, to not use them with Gen3?

Does romotecontrol definitely not use the EEPROM on GEN3, also?

Thanks a lot.

**Comment by martinpichlo** [comment]
Can anyone answer the question about the EEPROM?

**Reply by wills106** [reply]
It's not clear in the docs, but it indicates it still writes to the EEPROM on the Gen3 for part of the function.

The Gen4 approach would have to be rewritten, as the Gen3 doesn't have all of the options as per the Gen4.

There is a different approach on the Gen3 https://homeassistant-solax-modbus.readthedocs.io/en/latest/solax-G23-operation-modes/#grid-export-gen-3-only If you use positive values it will import from the Grid.

**Reply by martinpichlo** [reply]
Is there any way to check, if it's wrote to EEPROM? Is there e.g. a counter?

**Reply by wills106** [reply]
Not that I know off.

The Gen3 Hybrid docs mentions 1Million writes.
Probably more likely for the fans to pack in or a capacitor to fail before you kill the EEPROM?
